polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaSource.scala

```diff
@@ -339,20 +339,22 @@ class ScalaSource[G <: Global](
       case global.Select(qual: global.Tree, _) if qual != null =>
 
         // this brings in completions that are available through implicit enrichments
-        def addContext(scope: global.Scope): global.Scope = getContext(tree).map {
+        def addContext(scope: global.Scope): global.Scope = getContext(tree).flatMap {
           context =>
             val ownerTpe = qual.tpe match {
               case global.MethodType(List(), rtpe) => rtpe
               case _ => qual.tpe
             }
-            val allImplicits = new global.analyzer.ImplicitSearch(qual, global.definitions.functionType(List(ownerTpe), global.definitions.AnyTpe), isView = true, context0 = context).allImplicits
-            val implicitScope = scope.cloneScope
-            allImplicits.foreach {
-              result =>
-                val members = result.tree.tpe.finalResultType.members
-                members.foreach(implicitScope.enterIfNew)
-            }
-            implicitScope
+            if (ownerTpe != null) Either.catchNonFatal {
```

Comment:
Hmm, this is unrelated to coursier. Fixes a scary logged error when you edit the middle of a tree. Would be better if I could get that to type properly, but that's an issue for later.

Reply:
What's an example of editing the middle of a tree? What would you need to do to get this error?